packages/react-components/react-field/Spec.md

```diff
@@ -2,62 +2,338 @@
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+Each input component has a field version (such as `InputField`, `ComboboxField`, etc.) that includes the features of Field added to that component.
+
+```jsx
+<>
+  <InputField
+    // Field-specific props
+    label="This is the field label"
+    fieldOrientation="horizontal"
+    status="error"
+    statusText="This is error text"
+    // All props and slots of the underlying Input component are supported
+    required
+    size="small"
+    contentBefore="$"
+    contentAfter=".00"
+  />
+  <RadioGroupField label="Radio group field">
+    <Radio value="one" label="Option one" />
+    <Radio value="two" label="Option two" />
+    <Radio value="three" label="Option three" />
+  </RadioGroupField>
+  <ComboboxField label="Combobox field" status="success" statusText="Success text">
+    <Option value="one">Option one</Option>
+    <Option value="two">Option two</Option>
+    <Option value="three">Option three</Option>
+  </ComboboxField>
+  <SliderField label="Slider field" status="warning" statusText="Warning text" />
+  <SpinButtonField label="Spin button field" helperText="Help text" />
+</>
+```
+
+These field versions of the components use a common set of Field hooks, and can be defined using very little code.
+
+```ts
+export type InputFieldProps = FieldProps<typeof Input>;
+
+export const InputField: ForwardRefComponent<InputFieldProps> = React.forwardRef((props, ref) => {
+  const state = useField_unstable(props, ref, Input);
+  useFieldStyles_unstable(state);
+  return renderField_unstable(state);
+});
+
+InputField.displayName = 'InputField';
+```
+
+## Components
+
+The following field components will be defined. If more form components are added in the future, they should also include a Field version.
+
+- `CheckboxField`
+  - This will not use the Field's `label`, and instead forward the `label` prop to the underlying `Checkbox`.
+- `ComboboxField`
+- `DropdownField`
+- `InputField`
+- `RadioGroupField`
+- `SelectField`
+- `SliderField`
+- `SpinnerField`
+- `SwitchField`
+  - _Open question:_ how should this handle the `label`? Should it forward to the underlying `Switch`, or keep the label in the same place as the field's label? Might need a prop to control this behavior.
+- `TextareaField`
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+- **Orientation**: The `fieldOrientation` prop affects the layout of the label and field component:
+  - `'vertical'` (default) - label is above the field component
+  - `'horizontal'` - label is to the left of the field component, and is 33% the width of the field (this allows multiple stacked fields to all align their labels)
```

Comment:
For some reason when I wrote this I thought that Slider had an `orientation` prop that would conflict with this one, but it looks like it has a `vertical` prop instead. 

However, this could still result in a weird scenario like:
```jsx
<SliderField vertical orientation="horizontal" />
```

I did consider naming this `labelPosition?: 'above' | 'before'`. However, it affects the layout even when there is no label, and conflicts with the Checkbox and Switch `labelPosition` prop.

----

I'll go ahead and rename it to `orientation`, but if anyone has other ideas about what to call this, I'm open to suggestions.